crates/api/src/fetch.rs

```diff
@@ -120,6 +124,22 @@ pub trait FetchFactory: 'static + Send + Sync + std::fmt::Debug {
 /// Trait object [FetchFactory].
 pub type DynFetchFactory = Arc<dyn FetchFactory>;
 
+/// Summary of the fetch state.
+#[derive(Debug)]
+pub struct FetchStateSummary {
+    /// The op ids that are currently being fetched.
+    ///
+    /// Each op id is associated with one or more peer URL from which the op data could be
+    /// requested.
+    pub pending_requests: HashMap<OpId, Vec<Url>>,
+
+    /// The peer URL for nodes that are currently on backoff because of failed fetch requests.
```

Comment:
```suggestion
    /// The peer URL for nodes that are currently on backoff because of failed fetch requests, and the timestamp when that backoff will expire.
```